src/core/server/graph/extensions/LoggerExtension.ts

```diff
@@ -33,9 +39,36 @@ export function logQuery(
         : // A persisted query was not provided, parse the operation metadata
           // out from the document.
           getOperationMetadata(document)),
-    },
-    "graphql query"
-  );
+    };
+
+    // CORL-933: We are checking we have a logger as sometimes
+    // we are seeing no logger on the graph context. We're not
+    // sure why this is the case, but we want to catch when this
+    // is happening and not throw an error.
+    if (!ctx || !ctx.logger) {
+      // Catch the context and log it out, we need to know what is
+      // going on here
+      logger.warn({ ctx }, "unable to find logger on GraphContext");
+
+      // Continue to log the query.
+      // We don't want to lose the precious metrics.
+      logger.info(queryMeta, "graphql query");
+    } else {
+      // HAPPY STATE:
+      // Carry on as normal, everything is as it should be!
+      ctx.logger.info(queryMeta, "graphql query");
+    }
+  } catch (err) {
+    // Since we introduced all these checks, we don't know
+    // the state of the context or what other data may be
+    // undefined.
+    //
+    // If while compositing this log data we fail for other
+    // reasons, let's catch that to prevent other errors from
+    // occurring and log everything we can.
+    logger.error({ ctx, document }, "unable to log graphQL query");
+    logger.error(err);
```

Comment:
Same feedback from above, as well as noting that the document node is also pretty heavy..

Noting as well that calling `logger.error(err)` does not work, and instead you should use:

```ts
logger.error({ err }, "unable to log GraphQL query");
```

Note the `{ err }` as an object. This is required because bunyan (our logger) uses the key names to assist with serialization, and passing `logger.error(err)` provides no keys.